### PR TITLE
Add InfluxDbWriteable procedural derive macro

### PR DESCRIPTION
**Pull request by msrd0**
Fixes https://github.com/Empty2k12/influxdb-rust/issues/19

### CODE REVIEW

**Comment by msrd0** [comment]
The `InfluxDbWriteable` macro should now be working. I don't know if a `InfluxDbReadable` macro makes sense and what the use case would be? Maybe someone could give an example how they'd like to use it

**Comment by Empty2k12** [comment]
CI is failing with

> README.md out of date. Run "cargo readme > README.md" and commit again.

**Comment by msrd0** [comment]
@Empty2k12 I've fixed the readme. Kcov is still failing with an error message I cannot understand. Do you have an idea how to fix kcov?

**Comment by Empty2k12** [comment]
I would guess this is due to you changing the folder structure, resulting in a different folder structure inside the target folder which the kcov command relies on.

**Comment by Empty2k12** [comment]
As code coverage is inherently broken in Rust, even more so when using macros, I'd suggest removing code coverage in this PR and opening an issue to open it back up when better code coverage tools emerge or if a working solution has been found.
What do you think @msrd0 ?

File: influxdb/src/query/mod.rs [hunk 1]
```diff
@@ -87,21 +90,42 @@ pub trait Query {
...
+impl InfluxDbWriteable for Timestamp {
+    fn into_query(self, name: String) -> WriteQuery {
+        WriteQuery::new(self, name)
+    }
+}
+
 impl dyn Query {
     /// Returns a [`WriteQuery`](crate::WriteQuery) builder.
     ///
+    /// # Deprecated
```
Comment (Empty2k12) [inline influxdb/src/query/mod.rs]: Does it make sense to deprecate this vs remove it completely? I plan on moving the crate to 0.1 when this PR is merged and the initial stable version should be as clean as possible?

Comment (msrd0) [reply influxdb/src/query/mod.rs]: Well, if you want to bump to 0.1 it probably makes sense to remove deprecated items. If you want I can remove it.

Comment (Empty2k12) [reply influxdb/src/query/mod.rs]: I think I will do that in the Prepare Release Commit then

**Comment by kamolrat** [comment]
#23

**Comment by msrd0** [comment]
Keeping this up to date with master is getting annoying. What exactly is bloking this PR atm?

cc @Empty2k12

**Comment by Empty2k12** [comment]
Tests and CI will need to be looked at again.

I would suggest the following to get this merged soon:
 * Make any existing tests pass
 * remove code coverage step in this PR: it's quite broken for rust and does not work with your monorepo approach for the two crates.
 * document usage of the macro, as well as integrating a mention to this new documentation into the README.md (via app.rs and cargo-readme)

It might make sense to merge this before #41 so #41 can support writing Chrono dates as the timestamp.
 * create a different PR which changes CI to Github Actions

**Comment by msrd0** [comment]
> It might make sense to merge this before #41 so #41 can support writing Chrono dates as the timestamp.

#41 implements `From<DateTime<Utc>>` for `Timestamp`, so that should integrate without any issues.

File: influxdb/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+influxdb_derive = { version = "0.0.1", optional = true }
+reqwest = { version = "0.10.1", features = ["json"] }
+serde = { version = "1.0.104", features = ["derive"], optional = true }
+serde_json = { version = "1.0.46", optional = true }
+regex = "1.3.4"
+lazy_static = "1.4.0"
+
+# This is a temporary work around to fix a Failure-derive compilation error
+# Should be removed when https://github.com/Empty2k12/influxdb-rust/issues/48 is being done
+quote = "1.0.3"
```
Comment (Empty2k12) [inline influxdb/Cargo.toml]: This can now be removed.

File: influxdb/src/query/mod.rs [hunk 3]
```diff
@@ -129,33 +123,56 @@ pub trait Query {
...
+impl InfluxDbWriteable for Timestamp {
+    fn into_query<I: Into<String>>(self, name: I) -> WriteQuery {
+        WriteQuery::new(self, name.into())
+    }
+}
+
 impl dyn Query {
     /// Returns a [`WriteQuery`](crate::WriteQuery) builder.
     ///
+    /// # Deprecated
```
Comment (Empty2k12) [inline influxdb/src/query/mod.rs]: This should actually be removed now. I plan on releasing this PR as 0.1.0

**Comment by msrd0** [comment]
From my PoV, this PR is ready to be merged. Releasing will go just like you said ... release the derive crate first, wait a minute to give crates.io time to process the new/updated crate, and then release the main crate which has the derive crate as its dependency. However, note that there is no need to update the derive crate if no changes were made, just don't update the version either. It might be a good idea to keep major/minor versions in sync, at least that's what bigger libraries like diesel seem to be doing.

**Comment by Empty2k12** [comment]
Before releasing I will probably fix #54 to improve behavior on embedded platforms.

**Pull request merged**